### PR TITLE
added handler to add objective

### PR DESCRIPTION
**Pull request by DinAlla**

### CODE REVIEW

File: client/tests/src/redux/modules/employees/employeesActions.test.js [hunk 1]
```diff
@@ -109,20 +140,21 @@ describe('employees action', () => {
...
 
     test('should dispatch correct action in case of success response', () => {
       const dispatchSpy = jest.fn();
 
-      return actions.getObjectives()(dispatchSpy)
+      return actions.addObjective()(dispatchSpy)
         .then(() => {
-          const type = dispatchSpy.mock.calls[0][0].type;
+          const type = dispatchSpy.moc
+          k.calls[0][0].type;
```
Comment (TsvAliaksandra) [inline client/tests/src/redux/modules/employees/employeesActions.test.js]: ` k.calls[0][0].type;` ????
move in one line please

File: client/tests/src/redux/modules/employees/employeesReducer.test.js [hunk 2]
```diff
@@ -19,52 +19,72 @@ describe('employeesReducer', () => {
...
-        UnitManagerExternalId: '1',
-        InterviewDate: new Date(),
-        CreatedAt: new Date(),
-        UpdatedAt: new Date(),
-      },
-      employeeFullName: 'Alex tsvirko'
+      id: 1,
+      Archived: true,
+      EmployeeExternalId: '1',
+      UnitManagerExternalId: '1',
```
Comment (TsvAliaksandra) [inline client/tests/src/redux/modules/employees/employeesReducer.test.js]: i remember payload is not the same in store , there are 7 field, but in your case only 4
how it work???

File: client/tests/src/redux/modules/employees/employeesReducer.test.js [hunk 3]
```diff
@@ -19,52 +19,72 @@ describe('employeesReducer', () => {
...
-      StatusId: '1',
-      Progress: 1,
-      CreatedAt: new Date(),
-      UpdatedAt: new Date(),
+  test('Should return list of selected career day', async () => {
+    const payload = {
+      id: 1,
+      Archived: false,
+      EmployeeExternalId: '1',
+      UnitManagerExternalId: '1',
```
Comment (TsvAliaksandra) [inline client/tests/src/redux/modules/employees/employeesReducer.test.js]: the same

File: client/tests/src/redux/modules/employees/employeesActions.test.js [hunk 4]
```diff
@@ -82,13 +96,56 @@ describe('employees action', () => {
...
+
+  describe('method getSelectedCareerDay',() => {
+    const fakeResponse = {
+      body: {},
+      status: 500,
+    };
+    test('should sent GET request to `/api/objectives/1`',() => {
+      const dispatchSpy = jest.fn();
+
+      return actions.getSelectedCareerDay(1)(dispatchSpy)
```
Comment (TsvAliaksandra) [inline client/tests/src/redux/modules/employees/employeesActions.test.js]: move `1` in constant `careerDayId`

File: client/tests/src/redux/modules/employees/employeesActions.test.js [hunk 5]
```diff
@@ -58,10 +61,21 @@ describe('employees action', () => {
...
+      body: {},
+      status: 500,
+    };
     test('should sent GET request to `/api/career-days/1`', () => {
       const dispatchSpy = jest.fn();
-
-      return actions.getCareerDayOfEmployee()(dispatchSpy)
+      return actions.getCareerDayOfEmployee(employee)(dispatchSpy)
         .then(() => {
           const urlArg = axios.get.mock.calls[0][0];
```
Comment (TsvAliaksandra) [inline client/tests/src/redux/modules/employees/employeesActions.test.js]: rename urlArg to smth understandable

File: client/tests/src/redux/modules/employees/employeesActions.test.js [hunk 6]
```diff
@@ -98,33 +155,129 @@ describe('employees action', () => {
...
+      actions.getSelectedEmployee(employee)(dispatchSpy);
+      const returnedData = dispatchSpy.mock.calls[0][0].payload;
+
+      expect(returnedData).toEqual(employee);
+    });
+
+  });
+
+  describe('method addCareerDay',() => {
+    const careerDay = {
```
Comment (TsvAliaksandra) [inline client/tests/src/redux/modules/employees/employeesActions.test.js]: remove unusable constants

File: client/tests/src/redux/modules/employees/employeesActions.test.js [hunk 7]
```diff
@@ -98,33 +155,129 @@ describe('employees action', () => {
...
+      EmployeeExternalId: '1',
+      UnitManagerExternalId: 4,
+      InterviewDate: new Date(),
+    };
+    const fakeResponse = {
+      body: {},
+      status: 200,
+    };
+    const employee = {
```
Comment (TsvAliaksandra) [inline client/tests/src/redux/modules/employees/employeesActions.test.js]: the same

File: client/tests/src/redux/modules/employees/employeesActions.test.js [hunk 8]
```diff
@@ -98,33 +155,129 @@ describe('employees action', () => {
...
+        .then(() => expect(axios.post()).rejects.toEqual(fakeResponse));
+    });
+  });
+
+  describe('method addObjective', () => {
+    const objective = {
+      Title: 'objective',
+      Description: 'description',
+      CareerDayId: 1
+    }
```
Comment (TsvAliaksandra) [inline client/tests/src/redux/modules/employees/employeesActions.test.js]: missing semicolon

File: client/tests/src/redux/modules/employees/employeesActions.test.js [hunk 9]
```diff
@@ -1,10 +1,13 @@
-import EMPLOYEES_LIST from 'redux/modules/employees/employeesActionConstants';
-import * as actions from 'redux/modules/employees/employeesAction';
+import EMPLOYEES_LIST from 'redux/modules/employees/actionConstants';
+import * as actions from 'redux/modules/employees/actions';
 import axios from 'axios';
 
 describe('employees action', () => {
   beforeEach(() => {
     axios.get.mockReturnValue(Promise.resolve({status: 200, body: {}}));
+    axios.post = jest.fn((url) => {
```
Comment (TsvAliaksandra) [inline client/tests/src/redux/modules/employees/employeesActions.test.js]: `axios.post.mockReturnValue(Promise.resolve({status: 200, body: {}}));`

Comment (TsvAliaksandra) [reply client/tests/src/redux/modules/employees/employeesActions.test.js]: and in the axios.js file add post method

File: client/tests/src/redux/modules/employees/employeesActions.test.js [hunk 10]
```diff
@@ -1,10 +1,13 @@
...
+      return Promise.resolve({ body: {}, status: 200});
+  });
   });
```
Comment (TsvAliaksandra) [inline client/tests/src/redux/modules/employees/employeesActions.test.js]: use auto-alignment

File: client/tests/src/redux/modules/employees/employeesActions.test.js [hunk 11]
```diff
@@ -98,33 +155,129 @@ describe('employees action', () => {
...
+      Roles: '1',
+      LastName: 'Tsvirko',
+      FirstName: 'Alex',
+      PhotoUrl: 'my-avatar',
+      AccessToken: 'token',
+    };
+    
+    test('should dispatch correct action',() => {
+      const dispatchSpy = jest.fn();
+      actions.getSelectedEmployee()(dispatchSpy);
```
Comment (TsvAliaksandra) [inline client/tests/src/redux/modules/employees/employeesActions.test.js]: empty line

File: client/tests/src/redux/modules/employees/employeesActions.test.js [hunk 12]
```diff
@@ -98,33 +155,129 @@ describe('employees action', () => {
...
+      id: 1,
+      Roles: '1',
+      LastName: 'Tsvirko',
+      FirstName: 'Alex',
+      PhotoUrl: 'my-avatar',
+      AccessToken: 'token',
+    };
+    
+    test('should dispatch correct action',() => {
+      const dispatchSpy = jest.fn();
```
Comment (TsvAliaksandra) [inline client/tests/src/redux/modules/employees/employeesActions.test.js]: empty line

File: client/tests/src/redux/modules/employees/employeesActions.test.js [hunk 13]
```diff
@@ -98,33 +155,129 @@ describe('employees action', () => {
...
+      return actions.addCareerDay()(dispatchSpy)
+        .then(() => expect(axios.post()).rejects.toEqual(fakeResponse));
+    });
+  });
+
+  describe('method addObjective', () => {
+    const objective = {
+      Title: 'objective',
+      Description: 'description',
+      CareerDayId: 1
```
Comment (CeQadrat) [inline client/tests/src/redux/modules/employees/employeesActions.test.js]: missing traling comma

**Pull request merged**